holoviews/core/data/dictionary.py

```diff
@@ -213,7 +213,7 @@ def values(cls, dataset, dim, expanded=True, flat=True):
         if np.isscalar(values):
             if not expanded:
                 return np.array([values])
-            values = np.full(len(dataset), values)
+            values = np.full(len(dataset), values, dtype=np.array(values).dtype)
```

Comment:
This is the most bizarre bug I've come across in a long time. Before I added this ``dtype`` approach it would error (but only on Travis) saying that:

>ValueError: could not convert string to float: 'B'

But internally ``np.full`` does **exactly** the same thing I've done here: 

```python
def full(shape, fill_value, dtype=None, order='C'):
    if dtype is None:
        dtype = array(fill_value).dtype
    a = empty(shape, dtype, order)
    multiarray.copyto(a, fill_value, casting='unsafe')
    return a
```

Absolutely baffling, but it seems to work now.